rfc/rfc004-package-log.md

```diff
@@ -12,8 +12,21 @@ http://peter.bourgon.org/go-kit/#package-log
 
 ## Scope
 
-To be defined.
+* Log SHALL be drop in compatible with the standard library logger.
+* Log SHALL use key / value pairs for structured data.
+* Log MAY allow multiple stored logging contexts each with a different set of k/v pairs.
+* Log SHALL be transport agnostic by accepting and io.Writer as output target.
+* Log SHALL provide a default stdout io.Writer.
+* Log MAY implement io.MultiWriters allowing broadcast of logs over multiple transports.
+* Log MAY provide configurable back pressure handling strategies in the case of blocked io.Writers
+* Log MAY provide some built in transports such as syslog and logstash.
+* Log SHALL be format agnostic by providing an interface for log formatting.
+* Log MAY provide some built in formatters such as RFC3164, JSON, etc.
+* Log SHALL provide a set of defined severity levels.
```

Comment:
What do you think of a base logger interface, with only a single method for logging with fields, and then implementing different Warn/Error/etc. method-sets as wrapper types? Just a quick sketch, borrowing heavily from logrus:

```go
func (l *Logger) WithField(k string, v interface{}) *Entry
func (l *Logger) WithFields(Fields) *Entry

func (e *Entry) Printf(format string, args ...interface{})
```

```go
type BasicLogger struct{ *Logger }

func (l *BasicLogger) Printf(format string, args ...interface{}) {
    l.WithField("level", "default").Printf(format, args...)
}

func (l *BasicLogger) Errorf(format string, args ...interface{}) {
    l.WithField("level", "error").Printf(format, args...)
}
```

```go
type MyOrgLogger struct{ *Logger }

func (l *MyOrgLogger) Debugf(format string, args ...interface{}) {
    l.WithField("level", "debug").Printf(format, args...)
}

func (l *MyOrgLogger) Infof(format string, args ...interface{}) {
    l.WithField("level", "info").Printf(format, args...)
}

func (l *MyOrgLogger) Warnf(format string, args ...interface{}) {
    l.WithField("level", "warn").Printf(format, args...)
}

func (l *MyOrgLogger) Errorf(format string, args ...interface{}) {
    l.WithField("level", "error").Printf(format, args...)
}

func (l *MyOrgLogger) Fatalf(format string, args ...interface{}) {
    l.WithField("level", "fatal").Printf(format, args...)
    panic("kaboom")
}
```